sponsorship_compassion/models/move_line.py

```diff
@@ -35,16 +35,17 @@ def split_payment_and_reconcile(self):
                 move = line.move_id
                 move_line = line
                 break
-        else:
-            raise exceptions.UserError(
-                _(
+            else:
+                raise exceptions.UserError(
+                    _(
                     "This can only be done if one move line can be split "
                     "to cover the reconcile difference"
+                    )
                 )
-            )
 
         # Edit move in order to split payment into two move lines
         move.button_cancel()
+        move_line.payment_id.action_draft()
```

Comment:
Test and verify if this corrects CO-3613, and if we should restore the payment state after move.write or move.post.

Reply:
After some more talk and testing we confirmed that this is an issue and would leave the payment in draft state after the process.
The problem is that we cannot delete payments if they were already posted, and we cannot cancel them neither otherwise it deletes the associated move.

One solution would be to still cancel it and create another move replacing it. We must however test that this doesn't break the rest of the process.